homeassistant/helpers/check_config.py

```diff
@@ -102,7 +102,28 @@ def _pack_error(
 
     def _comp_error(ex: Exception, domain: str, config: ConfigType) -> None:
         """Handle errors from components: async_log_exception."""
-        result.add_warning(_format_config_error(ex, domain, config)[0], domain, config)
+        if domain in frontend_dependencies:
+            result.add_error(
+                _format_config_error(ex, domain, config)[0], domain, config
+            )
+        else:
+            result.add_warning(
+                _format_config_error(ex, domain, config)[0], domain, config
+            )
+
+    async def _get_integration(
+        hass: HomeAssistant, domain: str
+    ) -> loader.Integration | None:
+        """Get an integration."""
+        integration: loader.Integration | None = None
+        try:
+            integration = await async_get_integration_with_requirements(hass, domain)
+        except loader.IntegrationNotFound as ex:
```

Comment:
Can this only happen for custom integrations? Maybe add a comment?

Reply:
I think this is most likely to happen if the user has misspelled the name of the integration, it could also happen if a custom integration depends on other custom integrations which are not installed.

Reply:
Ok. I don't understand why we check for safe mode etc in this exception handler.

Reply:
Custom components are disabled in recovery mode and safe mode, so it might be confusing for the user if we complain about missing components in those modes.
Note that this is not changed in this PR.

In a follow-up, we should try to improve this so we know which custom components are available in safe mode, but without loading them.

Reply:
Yes, custom integrations was the only reason I knew that would make us check for safe mode and recovery mode. I think we should add a comment why we do that.